### PR TITLE
Added support for ZFS file system.

### PR DESCRIPTION
**Pull request by CrisBSD**
Hi to all qvd team, I maked the backend to support the zfs file system.

Best regards,

### CODE REVIEW

File: ext/QVD-HKD/Makefile.PL [hunk 1]
```diff
@@ -35,7 +35,8 @@ WriteMakefile(
         'Linux::Proc::Net::UDP' => 0,
         'Method::WeakCallback' => 0,
         'URI::Escape' => 0,
-        'BSD::Resource' => '1.2910'
+        'BSD::Resource' => '1.2910',
+	'Filesys::ZFS' => '1.5'
```
Comment (salva) [inline ext/QVD-HKD/Makefile.PL]: It seems that Filesys::ZFS is just a thin wrapper around `zpool` and `zfs` commands and you are not actually using more than its `_run` method. So, I think we can remove its usage completely from the module.

**Review by salva — CHANGES_REQUESTED** [review]

**Comment by salva** [comment]
@CrisBSD: I have created a new branch [zfs](/theqvd/theqvd/tree/zfs) with your code and some additional improvements. Could you check it?

I have changed the way the backend is configured. Now you only need to set `path.storage.basefs` and `path.storage.overlayfs` pointing to two paths corresponding to two ZFS datasets from a common pool. For instance:

    zpool create qvd /dev/sdb
    zfs create qvd/basefs
    zfs create qvd/overlayfs
    qa config set path.storage.basefs = /qvd/basefs
    qa config set path.storage.overlayfs = /qvd/overlayfs

**Review by salva — APPROVED** [review]

**Comment by CrisBSD** [comment]
@salva: I saw the code and think that is more clean and elegant, thanks for accept it. It would be great to integrate it into the master branch, and thank you for the learning, I hope send more contributions with this quality.

Best Regards,

**Comment by salva** [comment]
@CrisBSD, have you actually tried the version on the [zfs branch](https://github.com/theqvd/theqvd/tree/zfs)?

**Pull request closed without merging**